File: Data Structures & Algorithms/validate-parentheses/submission-1.cpp
#include <iostream>
#include <stack>
#include <unordered_map>
#include <string>
class Solution {
public:
    bool isValid(std::string s) {
        std::stack<char>mu_stack;
        std::unordered_map<char,char>pairs{
            {')','('},
            {']','['},
            {'}','{'}
        };

        for(char i : s){
            if (i=='('||i=='['||i=='{'){
                mu_stack.push(i);
            }
            else{
               
                if(mu_stack.empty()||mu_stack.top()!=pairs[i]){
                    return false;
                }
            
                mu_stack.pop();
                

    
            }
        }
        return mu_stack.empty();

        
    }
};

File: Data Structures & Algorithms/evaluate-reverse-polish-notation/submission-1.py
class Solution:
    def evalRPN(self, tokens: List[str]) -> int:
        operators = ["+", "-", "*", "/"]
        stack = []

        for n in tokens:
            if n in operators:
                b = stack.pop()
                a = stack.pop()

                if n == "+":
                    stack.append(a + b)
                elif n == "-":
                    stack.append(a - b)
                elif n == "*":
                    stack.append(a * b)
                elif n == "/":
                    stack.append(int(float(a) / b))  
            else:
                stack.append(int(n))

        return(stack[0])
        